app/models/manageiq/providers/ibm_terraform/inventory/parser/configuration_manager.rb

```diff
@@ -33,23 +35,33 @@ def configuration_profiles
   #         "18.215.xxx.yyy",
   #         "172.31.xx.yy"
   #     ],
+  #     "stacks": {
+  #         "templateId": "5ee8ddde6143d40017c2ee46"
+  #     },
   #     "tainted": false,
   #     "tenantId": "fbb9ab6a-b54d-43d4-9200-6d950700588f",
   #     "namespaceId": "acme",
   #     "namespaceMapping": true,
   #     "id": "5ee8ddde6143d40017c2ee45"
   # }]
-  def configured_systems
+  def configured_systems(persister_profiles)
+    persister_profiles_by_ref = persister_profiles.index_by do |profile|
+      profile.data[:manager_ref]
+    end
```

Comment:
If you're indexing by the manager_ref and you have all the data needed then you could just do a lazy_find instead